### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by SQLvariant**
Adding four functions to help automate deploying SSRS projects based on the deployment setting from the SSRS project file.

Fixes None.

Changes proposed in this pull request:
 - Creates two new 'helper' function, and two new functions for deployment
 - Creates a new Get-RsDeploymentConfig function for reading the SSRS project file
 - Creates a new Deploy-RsProject which deploys the SSRS project files based on the setting of the SSRS project file (which you can easily pipe in from the new Get-RsDeploymentConfig function)

How to test this code:
 - Take an SSRS 2019 project and deploy it to a Power BI Report Server
 - See examples in code.

Has been tested on (remove any that don't apply):
 - PowerShell 5.1
 - Windows 10
 - Power BI Report Server

### CODE REVIEW

File: ReportingServicesTools/Functions/CatalogItems/Deploy-RsProject.ps1 [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        [string]$OverwriteDataSources,
+        
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
+        [string]$FullPath,
+        
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+        [string]$ReportPortal
+        )
+
+Write-Host "`$RsProjectFolder = '$RsProjectFolder' being used is
```
Comment (jtarquino) [inline ReportingServicesTools/Functions/CatalogItems/Deploy-RsProject.ps1:79]: would you mind to fix the indentation

**Comment by jtarquino** [comment]
pelase remove the .suo file that is only for local configuration

**Comment by jtarquino** [comment]
Do you need the bin/debug folder of the project?

**Comment by jtarquino** [comment]
> CodeFlow for GitHub does not download executable files. If you trust this file and would like to run it, please download it yourself through GitHub.

do you need this .data file?

---
Refers to: Tests/CatalogItems/testResources/TestProjects/SQLServerPerformanceDashboardReportingSolution/SQL Server Performance Dashboard/database_storage_report.rdl.data:1 in d1279e2. [](commit_id = d1279e2c4f21821f9f3128c253bf6943c4012eba, deletion_comment = False)

**Comment by SQLvariant** [comment]
> > CodeFlow for GitHub does not download executable files. If you trust this file and would like to run it, please download it yourself through GitHub.
> 
> do you need this .data file?
> 
> Refers to: Tests/CatalogItems/testResources/TestProjects/SQLServerPerformanceDashboardReportingSolution/SQL Server Performance Dashboard/database_storage_report.rdl.data:1 in [d1279e2](https://github.com/microsoft/ReportingServicesTools/commit/d1279e2c4f21821f9f3128c253bf6943c4012eba). [](commit_id = [d1279e2](https://github.com/microsoft/ReportingServicesTools/commit/d1279e2c4f21821f9f3128c253bf6943c4012eba), deletion_comment = False)

If I do need it, I guess we'll find out when the build fails 🙂

File: ReportingServicesTools/Functions/CatalogItems/Publish-RsProject.ps1 [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            Specify the location of the SSRS project file whose deployment profiles should be fetched.
+
+        .EXAMPLE
+            Get-RsDeploymentConfig -ProjectFile 'C:\Users\Aaron\source\repos\Finance\Financial Reports\SSRS_FR\SSRS_FR.rptproj' |
+            Add-Member -PassThru -MemberType NoteProperty -Name ReportPortal -Value 'http://localhost/PBIRSportal/' |
+            Publish-RsProject
+
+            Description
+            -----------
+            Get-RsDeploymentConfig prompts the user to select whick deployment configuration to use from
```
Comment (jtarquino) [inline ReportingServicesTools/Functions/CatalogItems/Publish-RsProject.ps1:20]: typo which

File: ReportingServicesTools/Functions/CatalogItems/Get-RsDeploymentConfig.ps1 [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+if($ConfigurationToUse){
+    [XML]$rptproj = Get-Content $RsProjectFile
+    $Deployment = $rptproj.Project.PropertyGroup | where { $_.FullPath -eq $ConfigurationToUse }
+
+        $RSConfig = [pscustomobject]@{
+            FullPath               = $Deployment.FullPath
+            OverwriteDatasets      = $Deployment.OverwriteDatasets
+            OverwriteDataSources   = $Deployment.OverwriteDataSources
+            TargetReportFolder     = (StripTrailingSlash $Deployment.TargetReportFolder)
```
[the remark this answers is not in the document]
Comment (SQLvariant) [reply ReportingServicesTools/Functions/CatalogItems/Get-RsDeploymentConfig.ps1]: Looks like `.Trimend("/")` works best here.

**Pull request merged**